docs/holiday_categories.md

```diff
@@ -0,0 +1,322 @@
+# Holiday Categories
+
+The holidays library supports various categories of holidays to help classify different types of observances. This allows users to filter holidays based on their official status, religious significance, or cultural importance.
+
+## Overview
+
+Holiday categories enable you to filter holidays based on their nature and official recognition. Each country defines which categories it supports, and holidays are classified accordingly.
+
+## Category Types
+
+### Official Status Categories
+
+**PUBLIC**
+Official holidays recognized by the government that typically provide time off from work and school for the general population.
+
+Examples:
+- New Year's Day in most countries
+- Independence Day in the United States
+- Christmas Day in Christian-majority countries
+
+**GOVERNMENT**
+Holidays observed by government institutions but may not provide general time off for all workers.
+
+Examples:
+- Flag Day in some jurisdictions
+- Government-specific commemorative days
+
+**WORKDAY**
+Holidays that are officially recognized but do not provide time off from work. Often ceremonial or commemorative in nature.
+
+Examples:
+- Constitution Day in some countries
+- Memorial days that are observed but not as public holidays
+
+**UNOFFICIAL**
+Commonly celebrated cultural holidays with no official government recognition or mandated time off. These are widely observed cultural celebrations, but exclude novelty, internet, or marketing "national days."
+
+Examples:
+- Valentine's Day
+- Saint Patrick's Day  
+- Halloween
+
+### Institutional Categories
+
+**BANK**
+Holidays specifically observed by banking institutions, which may differ from general public holidays.
+
+Examples:
+- Good Friday (in some countries)
+- Additional banking-specific closures
+
+**SCHOOL**
+Holidays specific to educational institutions.
+
+Examples:
+- Teacher's Day
+- Education-specific observances
+
+**ARMED_FORCES**
+Holidays specific to military personnel and institutions.
+
+Examples:
+- Armed Forces Day
+- Military commemoration days
+
+### Special Categories
+
+**OPTIONAL**
+Holidays that individuals or organizations may choose to observe, often with flexibility in implementation.
+
+Examples:
+- Religious holidays for minority populations
+- Cultural observances with optional recognition
+
+**MANDATORY**
+Holidays that must be observed by law in specific contexts.
```

Comment:
AFAIK `MANDATORY` is currently only used by Macau at the moment for its Statutory Holidays that must be day-off i.e. https://www.dsal.gov.mo/en/standard/holiday_table.html (its public list currently represents its "General" Holidays list.

@KJhellico @arkid15r Do we want to phase out this particular category and move the "General" list to `OPTIONAL` category instead, like Hong Kong's implementation?

Reply:
Yes, I think it will be better. MANDATORY name doesn't look intuitive. :)